hazelcast/src/main/java/com/hazelcast/monitor/impl/LocalIndexStatsImpl.java

```diff
@@ -0,0 +1,303 @@
+        this.totalRemoveLatency = totalRemoveLatency;
+    }
+
+    @Override
+    public long getMemoryCost() {
+        return memoryCost;
+    }
+
+    /**
+     * Sets the memory cost of this stats to the given memory cost value.
+     *
+     * @param memoryCost the memory cost value to set.
+     */
+    public void setMemoryCost(long memoryCost) {
+        this.memoryCost = memoryCost;
+    }
+
+    /**
+     * Sets all the values in this stats to the corresponding values in the
+     * given on-demand stats.
+     *
+     * @param onDemandStats the on-demand stats to fetch the values to set from.
+     */
+    public void setAllFrom(OnDemandIndexStats onDemandStats) {
+        this.creationTime = onDemandStats.getCreationTime();
+        this.hitCount = onDemandStats.getHitCount();
+        this.queryCount = onDemandStats.getQueryCount();
+        this.averageHitSelectivity = onDemandStats.getAverageHitSelectivity();
+        this.averageHitLatency = onDemandStats.getAverageHitLatency();
+        this.insertCount = onDemandStats.getInsertCount();
+        this.totalInsertLatency = onDemandStats.getTotalInsertLatency();
+        this.updateCount = onDemandStats.getUpdateCount();
+        this.totalUpdateLatency = onDemandStats.getTotalUpdateLatency();
+        this.removeCount = onDemandStats.getRemoveCount();
+        this.totalRemoveLatency = onDemandStats.getTotalRemoveLatency();
+        this.memoryCost = onDemandStats.getMemoryCost();
+    }
+
+    @Override
+    public JsonObject toJson() {
+        JsonObject json = new JsonObject();
+        json.add("creationTime", creationTime);
+        json.add("hitCount", hitCount);
+        json.add("queryCount", queryCount);
+        json.add("averageHitSelectivity", averageHitSelectivity);
+        json.add("averageHitLatency", averageHitLatency);
+        json.add("insertCount", insertCount);
+        json.add("totalInsertLatency", totalInsertLatency);
+        json.add("updateCount", updateCount);
+        json.add("totalUpdateLatency", totalUpdateLatency);
+        json.add("removeCount", removeCount);
+        json.add("totalRemoveLatency", totalRemoveLatency);
+        json.add("memoryCost", memoryCost);
+        return json;
+    }
+
+    @Override
+    public void fromJson(JsonObject json) {
+        creationTime = json.getLong("creationTime", -1);
+        hitCount = json.getLong("hitCount", -1);
+        queryCount = json.getLong("queryCount", -1);
+        averageHitSelectivity = json.getDouble("averageHitSelectivity", -1.0);
+        averageHitLatency = json.getLong("averageHitLatency", -1);
+        insertCount = json.getLong("insertCount", -1);
+        totalInsertLatency = json.getLong("totalInsertLatency", -1);
+        updateCount = json.getLong("updateCount", -1);
+        totalUpdateLatency = json.getLong("totalUpdateLatency", -1);
+        removeCount = json.getLong("removeCount", -1);
+        totalRemoveLatency = json.getLong("totalRemoveLatency", -1);
```

Comment:
can we see this as an inconsistency compared to the values returned from `InternalIndexStats#EMPTY`? Here default values are `-1` instead of `0`?

Reply:
Those `-1`s are just placeholders for missing values, which should not be missing under normal circumstances. This technique is used by all stats implementing `JsonSerializable`, I suspect MC may depend on that. `InternalIndexStats` is never exposed to a user, so there is no difference what values are considered "empty" by it internally.